### PR TITLE
Add ability to read secrets in parallel

### PR DESCRIPTION
**Pull request by RyanGordon**
User story: Sometimes there can be a lot of secrets in a single directory. This makes daytona take 30-60 seconds for every container that spins up that runs daytona on startup. I'd like to for daytona to take less time in these situations.

Solution: This change implements a worker pool for calling the vault read API. Per https://github.com/hashicorp/vault/issues/272#issuecomment-373178001 the suggested solution is to make http requests in parallel.

This PR adds a new option that defaults to 1 worker and is limited to 5 workers so as to not allow a crazy high number of parallel workers.

Before: 30-60 seconds to read ~100 secrets
After: 3-15 seconds to read ~100 secrets @ 5 workers

### CODE REVIEW

File: cmd/daytona/main.go [hunk 1]
```diff
@@ -107,6 +108,19 @@ func init() {
...
 	flag.StringVar(&config.AuthMount, "auth-mount", cfg.BuildDefaultConfigItem("AUTH_MOUNT", ""), "")
+	flag.IntVar(&config.NumSecretReadWorkers, "num-secret-read-workers", func() int {
+		b, err := strconv.ParseInt(cfg.BuildDefaultConfigItem("NUM_SECRET_READ_WORKERS", "1"), 10, 64)
+		if err != nil {
+			return 1
+		}
+		if b < 1 {
+			b = 1
+		}
+		if b > 5 {
```
Comment (robison) [inline cmd/daytona/main.go]: Would consider just returning an error instead of silently lowering the value

File: cmd/daytona/main.go [hunk 2]
```diff
@@ -107,6 +108,19 @@ func init() {
 	flag.StringVar(&config.AWSAuthMount, "iam-auth-mount", cfg.BuildDefaultConfigItem("IAM_AUTH_MOUNT", "aws"), "the vault mount where iam auth takes place (env: IAM_AUTH_MOUNT)")
 	flag.StringVar(&config.GCPAuthMount, "gcp-auth-mount", cfg.BuildDefaultConfigItem("GCP_AUTH_MOUNT", "gcp"), "the vault mount where gcp auth takes place (env: GCP_AUTH_MOUNT)")
 	flag.StringVar(&config.AuthMount, "auth-mount", cfg.BuildDefaultConfigItem("AUTH_MOUNT", ""), "")
+	flag.IntVar(&config.NumSecretReadWorkers, "num-secret-read-workers", func() int {
```
Comment (robison) [inline cmd/daytona/main.go]: Nit: shorter varname, `-workers` or something similar?

File: pkg/secrets/reader.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+		case keyPath := <-pr.keyPathInChan:
+			secret, err := pr.client.Logical().Read(keyPath)
+			pr.secretOutChan <- &SecretResult{
+				KeyPath: keyPath,
+				Secret:  secret,
+				Err:     err,
+			}
+		}
+	}
+}
```
Comment (robison) [inline pkg/secrets/reader.go]: No test coverage on reader.go - would be nice to mock/test at least some of it.

**Comment by RyanGordon** [comment]
@robison Thanks for the review! I addressed your feedback and additionally added mocked tests around `reader.go`

**Pull request merged**